pkg/plugins/policies/meshhttproute/api/v1alpha1/meshhttproute.go

```diff
@@ -41,11 +41,26 @@ type Rule struct {
 }
 
 type Match struct {
-	Path PathMatch `json:"path,omitempty"`
+	Path   *PathMatch `json:"path,omitempty"`
+	Method *Method    `json:"method,omitempty"`
 }
 
+// +kubebuilder:validation:Enum=Exact;Prefix;RegularExpression
+type PathMatchType string
+
+// +kubebuilder:validation:Enum=CONNECT;DELETE;GET;HEAD;OPTIONS;PATCH;POST;PUT;TRACE
+type Method string
+
+const (
+	Exact             PathMatchType = "Exact"
+	Prefix            PathMatchType = "Prefix"
+	RegularExpression PathMatchType = "RegularExpression"
+)
+
 type PathMatch struct {
-	Prefix string `json:"prefix,omitempty"`
+	// +kubebuilder:validation:MinLength=1
+	Value string        `json:"value"`
+	Type  PathMatchType `json:"type"`
```

Comment:
docs plz. I'd especially add a comment about abolsute path requirement

Reply:
done. also fixed the semantics for prefix matches similar to how it's handled in gateway

Reply:
Going to merge this. Let me know if something doesn't look right!